web/app/scripts/views/map/layers-controller.js

```diff
@@ -322,35 +320,38 @@
                 blackspotUrl = TileUrlService.blackspotsUrl(uuid);
                 blackspotUtfUrl = TileUrlService.blackspotsUtfGridUrl(uuid);
             }
+            var secondaryRecordsUrl = $q.when('');
+            var secondaryUtfGridUrl = $q.when('');
+            if (ctl.secondaryType) {
+                secondaryRecordsUrl = TileUrlService.secondaryTilesUrl(ctl.secondaryType);
+                secondaryUtfGridUrl = TileUrlService.recUtfGridTilesUrl(ctl.secondaryType);
+            }
+
             return $q.all(
-                [
-                    TileUrlService.recTilesUrl(ctl.recordType),
-                    TileUrlService.recUtfGridTilesUrl(ctl.recordType),
-                    TileUrlService.recHeatmapUrl(ctl.recordType),
-                    blackspotUrl,
-                    blackspotUtfUrl,
-                    blackspotTileKey
-                ]
+                {
+                    baseRecordsUrl:       TileUrlService.recTilesUrl(ctl.recordType),
+                    baseUtfGridUrl:       TileUrlService.recUtfGridTilesUrl(ctl.recordType),
+                    baseHeatmapUrl:       TileUrlService.recHeatmapUrl(ctl.recordType),
+                    blackspotsUrl:        blackspotUrl,
+                    blackspotsUtfGridUrl: blackspotUtfUrl,
+                    blackspotTileKey:     blackspotTileKey,
+                    secondaryRecordsUrl:  secondaryRecordsUrl,
+                    secondaryUtfGridUrl:  secondaryUtfGridUrl,
+                }
             );
         }
 
         /**
-         * Given an array of urls in form:
-         * [baseRecordsUrl, baseUtfGridUrl, baseHeatmapUrl, blackspotsUrl],
+         * Given the urls object returned by getUrls,
          * updates the layer groups so that the layer switcher does not
          * need to be re-initialized every time layer urls change
          */
         function updateLayerGroups(urls) {
-            var baseRecordsUrl = urls[0];
-            var baseUtfGridUrl = urls[1];
-            var baseHeatmapUrl = urls[2];
-            var blackspotsUrl = urls[3];
-            var blackspotsUtfGridUrl = urls[4];
-            var blackspotTileKey = urls[5];
-
-            updateEventLayer(baseRecordsUrl, baseUtfGridUrl);
-            updateHeatmapLayer(baseHeatmapUrl);
-            updateBlackspotLayer(blackspotsUrl, blackspotsUtfGridUrl, blackspotTileKey);
+            updateEventLayer(urls.baseRecordsUrl, urls.baseUtfGridUrl);
+            updateSecondaryLayer(urls.secondaryRecordsUrl, urls.secondaryUtfGridUrl);
```

Comment:
Yeah, it's a little muddy.  The accidents layer goes by many names--base, event, primary, and that's not to mention the places where it's implicit (e.g. recordType).  Actually, I'm going to do a pass of standardizing that, at least within LayersController.

I think it works, though, for what it is now.  If we wanted to make the primary and/or secondary record types switchable, we'd want to clean up the nomenclature in more places and I think we'd find some spots where hidden assumptions or hard-coding would need to be refactored out.  If we wanted to go further and support an arbitrary number of active record types, that would be another matter.

But yeah, my feeling is at this point the level of abstraction is not too cumbersome and is providing some advantages.